master/internal/experiment.go

```diff
@@ -275,11 +281,15 @@ func (e *experiment) Receive(ctx *actor.Context) error {
 	case model.State:
 		e.updateState(ctx, msg)
 	case sproto.SetGroupMaxSlots:
-		e.Config.Resources.MaxSlots = msg.MaxSlots
+		resources := e.Config.Resources()
+		resources.SetMaxSlots(msg.MaxSlots)
+		e.Config.SetResources(resources)
```

Comment:
Because pointers are nilable and nil means "not present".  Arguably, if `e.Config.Resources()` returned a pointer that was guaranteed-not-nil that would be similar in safety to returning a struct, but only because we know ahead of  time that `e.Config.Resources()` will always have some default values.

The current Getter APIs are 100% consistent in that any getter that returns a pointer might be nil, and you must check for nil values at runtime.  IMO it would be obnoxious if some getters returned pointers only to make them modifiable.

Also... I didn't feel like the experiment config needed to be optimized for updates.  We should really not be updating it.  We should be pulling parts out of it that the system cares about (like slots_per_trial, image, podspec, etc) and holding on to those in some system-specific format that doesn't need any of the expconf framework.  And the experiment config should be just that: how the user configured the experiment; immutable after experiment submission.

We pack way too much system-level stuff into the experiment config so that we can use the experiment config as an internally-useful object.  A glaring example of this is `ExperimentConfig.Internal`.  If you git-blame it far enough, you'll see my name on it.  @ddn0 told me, "hey, you know normally you don't inject this kind of information into the user's config, you hold on to it on the side", but I didn't understand what he meant by that when he told me.

Reply:
> We pack way too much system-level stuff into the experiment config so that we can use the experiment config as an internally-useful object. A glaring example of this is ExperimentConfig.Internal. If you git-blame it far enough, you'll see my name on it. @ddn0 told me, "hey, you know normally you don't inject this kind of information into the user's config, you hold on to it on the side", but I didn't understand what he meant by that when he told me.

Yeah.. I've felt this for a long time.